client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@angular/core';
+import { FileExportService } from './file-export.service';
+import { TranslateService } from '@ngx-translate/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class CsvExportService {
+    /**
+     * Constructor
+     */
+    public constructor(protected exporter: FileExportService, private translate: TranslateService) {}
+
+    /**
+     * Saves an array of model data to a CSV.
+     * @param data Array of Model instances to be saved
+     * @param columns Column definitions
+     * @param filename name of the resulting file
+     * @param options optional:
+     *      lineSeparator (defaults to \r\n windows style line separator),
+     *      columnseparator defaults to semicolon (other usual separators are ',' '\T' (tab), ' 'whitespace)
+     */
+    public export<T extends BaseViewModel>(
+        data: T[],
+        columns: {
+            property: keyof T; // name of the property used for export
+            label?: string;
+            assemble?: string; // (if property is further child object, the property of these to be used)
+        }[],
+        filename: string,
+        { lineSeparator = '\r\n', columnSeparator = ';' }: { lineSeparator?: string; columnSeparator?: string } = {}
+    ): void {
+        const allLines = []; // Array of arrays of entries
+        const usedColumns = []; // mapped properties to be included
+
+        // initial array of usable text separators. The first character not used
+        // in any text data or as column separator will be used as text separator
+        let tsList = ['"', "'", '`', '/', '\\', ';', '.', ','];
+
+        if (lineSeparator === columnSeparator) {
+            throw new Error('lineseparator and columnseparator must differ from each other');
+        }
+
+        tsList = this.checkCsvTextSafety(lineSeparator, tsList);
+        tsList = this.checkCsvTextSafety(columnSeparator, tsList);
+
+        // create header data
+        const header = [];
+        columns.forEach(column => {
+            const rawLabel: string = column.label ? column.label : (column.property as string);
+            const colLabel = this.capitalizeTranslate(rawLabel);
+            tsList = this.checkCsvTextSafety(colLabel, tsList);
+            header.push(colLabel);
+            usedColumns.push(column.property);
+        });
+        allLines.push(header);
+        // create lines
+        data.forEach(item => {
+            const line = [];
+            for (let i = 0; i < usedColumns.length; i++ ){
+                const property = usedColumns[i];
+                let prop: any = item[property];
+                if (columns[i].assemble){
+                    const subproperties = [];
+                    item[property].forEach(subitem => {
+                        subproperties.push(subitem[columns[i].assemble]);
+                    });
+                    prop = subproperties.join(' ');
```

Comment:
`prop = item[property].map(subitem => this.translate.instant(subitem[columns[i].assemble])).join(',');`
Also remove the `,` from tsList.

Reply:
So, the assumptions are: The data of the assembled subitems are always strings to be translated, and they will be comma separated. Well then.

Reply:
yes. Maybe ad a toString then; but we can change this, if it is the case.